projects/framework-houdini/source/ftrack_framework_houdini/__init__.py

```diff
@@ -0,0 +1,206 @@
+        f' {framework_extensions_path}'
+    )
+
+    global client_instance
+
+    # Create ftrack session and instantiate event manager
+    session = ftrack_api.Session(auto_connect_event_hub=False)
+    event_manager = EventManager(
+        session=session, mode=constants.event.LOCAL_EVENT_MODE
+    )
+    logger.debug(f"framework_extensions_path:{framework_extensions_path}")
+    # Instantiate registry
+    registry_instance = Registry()
+    registry_instance.scan_extensions(paths=framework_extensions_path)
+
+    # TODO: clean up this dictionary creation or move it as a query function of
+    #  the registry.
+    # Create a registry dictionary with all extension names to pass to the mix panel event
+    registry_info_dict = {
+        'tool_configs': [
+            item['name'] for item in registry_instance.tool_configs
+        ]
+        if registry_instance.tool_configs
+        else [],
+        'plugins': [item['name'] for item in registry_instance.plugins]
+        if registry_instance.plugins
+        else [],
+        'engines': [item['name'] for item in registry_instance.engines]
+        if registry_instance.engines
+        else [],
+        'widgets': [item['name'] for item in registry_instance.widgets]
+        if registry_instance.widgets
+        else [],
+        'dialogs': [item['name'] for item in registry_instance.dialogs]
+        if registry_instance.dialogs
+        else [],
+        'launch_configs': [
+            item['name'] for item in registry_instance.launch_configs
+        ]
+        if registry_instance.launch_configs
+        else [],
+        'dcc_configs': [item['name'] for item in registry_instance.dcc_configs]
+        if registry_instance.dcc_configs
+        else [],
+    }
+
+    # Set mix panel event
+    set_usage_tracker(
+        UsageTracker(
+            session=session,
+            default_data=dict(
+                app="Houdini",
+                registry=registry_info_dict,
+                version=__version__,
+                app_version=hou.applicationVersionString(),
+                os=platform.platform(),
+            ),
+        )
+    )
+
+    # Instantiate Host and Client
+    Host(event_manager, registry=registry_instance)
+    client_instance = Client(event_manager, registry=registry_instance)
+
+    # Init tools
+    dcc_config = registry_instance.get_one(
+        name='framework-houdini', extension_type='dcc_config'
+    )['extension']
+
+    logger.debug(f'Read DCC config: {dcc_config}')
+
+    # Generate ftrack menu
+    root, ftrack_menu = get_ftrack_menu()
+
+    for tool in dcc_config['tools']:
+        menu_item = ET.SubElement(ftrack_menu, "scriptItem")
+        menu_item.set("id", tool['name'])
+        label = ET.SubElement(menu_item, "label")
+        label.text = tool['label']
+        menu_item_script = ET.SubElement(menu_item, "scriptCode")
+        menu_item_script.text = _get_menu_item_script(tool)
```

Comment:
why setting the client_instance as a global variable? can you pass the client_instance as an argument here as well?

Reply:
I think this is the correct way to do it, we have the same approach with the most recent bootstrap implementation (compare to Nuke: https://github.com/ftrackhq/integrations/blob/2ac705aebddc586ca8f792c15d0638e9568847f4/projects/framework-nuke/source/ftrack_framework_nuke/__init__.py#L89)

Reply:
Nuke needs it as global because is not called directly, but not this one. But no problem its ok for me to keep it.

Reply:
Similar reason here, I don't remember from the top of my head but I think it was related to the fact that the call had to be written to a XML file. Still ok to keep as it is for now?